test/swar/BasicOperations.cpp

```diff
@@ -323,3 +337,76 @@ constexpr auto aBooleansWithTrue = booleans(SWAR<4, u32>{0x1});
 static_assert(aBooleansWithTrue);
 static_assert(!aBooleansWithTrue); // this is a pitfall, but lesser evil?
 static_assert(false == !bool(aBooleansWithTrue));
+
+TEST_CASE(
+    "fullAddition",
+    "[swar]"
+) {
+    {
+        // 0x0111 (7) + 0x0001 (1) is 0x1000 (-8)
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'1000), SWAR<4, u32>(0x0000'7000));
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.overflow.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.result.value());
+    }
+    {
+        // 0x0111 (7) + 0x0001 (1) is 0x1000 (-8)
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'8000), SWAR<4, u32>(0x0000'7000));
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.overflow.value());
+        CHECK(SWAR<4, u32>(0x0000'F000).value() == fplusone.result.value());
+    }
+    {
+        // 0x0101 (5) + 0x0101 (5) is 0x1010 (-x)
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'5000), SWAR<4, u32>(0x0000'5000));
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.overflow.value());
+        CHECK(SWAR<4, u32>(0x0000'A000).value() == fplusone.result.value());
+    }
+    {
+        // 0x0111 (7) + 0x0111 (7) is 0x1110 (0x1110->0x1101->0x0010) (e unsigned, 2 signed)
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'7000), SWAR<4, u32>(0x0000'7000));
+        CHECK(SWAR<4, u32>(0x0000'0000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.overflow.value());
+        CHECK(SWAR<4, u32>(0x0000'e000).value() == fplusone.result.value());
+    }
+    {
+        const auto fplusone = fullAddition(SWAR<4, u32>(0x0000'a000), SWAR<4, u32>(0x0000'a000));
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.carry.value());
+        CHECK(SWAR<4, u32>(0x0000'8000).value() == fplusone.overflow.value());
+    }
+}
+
+TEST_CASE(
+    "booleanSWARasMask",
```

Comment:
Going to the name, I don't think `asMask` is the right name, since just by themselves, the boolean swars are masks, they're used as masks in many places.
What you mean by a mask here is the mask of the whole lane or swar element.

Reply:
resolved elsewhere